File: event/services/algolia.go
package services

import (
	"fmt"

	"github.com/algolia/algoliasearch-client-go/algoliasearch"
	"github.com/ninjadotorg/handshake-cryptosign/event/config"
	"github.com/ninjadotorg/handshake-cryptosign/event/models"
)

// AlgoliaService : struct
type AlgoliaService struct{}

// Result : struct
type Result struct {
	ObjectID     int
	MatchName    string
	SourceName   string
	SourceURL    string
	CategoryName string
}

// ImportMatches : matches
func (s AlgoliaService) ImportMatches(matches []models.Match) (err error) {
	conf := config.GetConfig()

	algoliaData := conf.GetStringMapString("algolia")
	var algoliaApplicationID, algoliaAPIKey, indexName string
	for key, value := range algoliaData {
		if key == "application_id" {
			algoliaApplicationID = value
		} else if key == "api_key" {
			algoliaAPIKey = value
		} else if key == "index_name" {
			indexName = value
		}
	}

	client := algoliasearch.NewClient(algoliaApplicationID, algoliaAPIKey)
	var objects []algoliasearch.Object

	for index := 0; index < len(matches); index++ {
		m := matches[index]
		data := make(map[string]interface{})
		data["objectID"] = m.ID
		data["eventName"] = m.Name
		data["sourceName"] = m.Source.Name
		data["sourceURL"] = m.Source.URL
		objects = append(objects, data)
	}

	index := client.InitIndex(indexName)
	settings := algoliasearch.Map{
		"searchableAttributes": []string{
			"eventName",
			"sourceURL",
			"sourceName",
		},
	}
	index.SetSettings(settings)
	res, err := index.AddObjects(objects)
	fmt.Println(res)
	return err
}

File: restapi/app/extensions/algolia.py
from algoliasearch import algoliasearch

class Algolia(object):
	def __init__(self, app=None):
		super(Algolia, self).__init__()
		if app:
			self.app = app
			self.load_algolia()


	def init_app(self, app):
		self.app = app
		self.load_algolia()


	def load_algolia(self):
		client = algoliasearch.Client(self.app.config['ALGOLIA_APPLICATION_ID'], self.app.config['ALGOLIA_API_KEY'])
		self.index = client.init_index(self.app.config['ALGOLIA_INDEX_NAME'])


	def search(self, text):
		return self.index.search(text)
